server/xcode/xcode_darwin.go

```diff
@@ -41,27 +43,20 @@ func NewXcodeLocator() (*xcodeLocator, error) {
 	return xl, err
 }
 
-// Finds the Xcode developer directory that most closely matches the given Xcode version.
-func (x *xcodeLocator) DeveloperDirForVersion(version string) (string, error) {
-	xv := x.xcodeVersionForVersionString(version)
+// Finds the Xcode that most closely matches the given Xcode version.
+// Returns the developer directory for that Xcode and the SDK root for the given SDK.
+func (x *xcodeLocator) PathsForVersionAndSDK(xcodeVersion string, sdk string) (string, string, error) {
+	xv := x.xcodeVersionForVersionString(xcodeVersion)
 	if xv == nil {
-		return "", status.FailedPreconditionErrorf("Xcode version %s not installed on remote executor. Available Xcode versions are %+v", version, x.versions)
-	}
-	return xv.developerDirPath, nil
-}
 
-// Return true if the given SDK path is present in the given Xcode version.
-func (x *xcodeLocator) IsSDKPathPresentForVersion(sdkPath, version string) bool {
-	xv := x.xcodeVersionForVersionString(version)
-	if xv == nil {
-		return false
+		return "", "", status.FailedPreconditionErrorf("Xcode version %s not installed on remote executor. Available Xcode versions are %+v", xcodeVersion, x.versions)
```

Comment:
ubernit, but can we make a nicer string instead of printing the map? (Since this is for users...)

func versionsString(versions map[string]*xcodeVersion) string {
  availableVersionsList := make([]string, 0, len(x.versions))
  for version, _ := range x.versions {
    availableVersions = append(availableVersions, version)
  }
  sort.Strings(availableVersionsList)
  return strings.Join(availableVersionsList, ", ")
}
...
status.FailedPreconditionErrorf("Xcode version %s not installed on remote executor. Available Xcode versions are %s", xcodeVersion, versionsString(x.versions))